### ISSUE TITLE
ALB Ingress Weighted Target Group Support

### ISSUE THREAD

**Issue by jessesuen** [argoproj/argo-rollouts#421]
ALB Ingress Controller recently added support for [AWS Weighted Target Groups](https://aws.amazon.com/blogs/aws/new-application-load-balancer-simplifies-deployment-with-weighted-target-groups/) as of their most recent [v1.1.5 release](https://github.com/kubernetes-sigs/aws-alb-ingress-controller/releases/tag/v1.1.5).

https://github.com/kubernetes-sigs/aws-alb-ingress-controller/pull/1088

The ingress annotation for using weighted target groups look like:

```yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: simple-ingress-example
  annotations:
    alb.ingress.kubernetes.io/actions.weighted-routing: '{"Type":"forward","ForwardConfig":{"TargetGroupStickinessConfig":{"Enabled":"True","DurationSeconds":100},"TargetGroups":[{"Weight":20,"ServiceName":"Service1","ServicePort":"80"},{"Weight":80,"TargetGroupArn":"arn:aws:******/external-legecy-tg"}]}}'
spec:
  rules:
  - host: m00nf1sh.example.com
    http:
      paths:
      - path: /foo
        backend:
          serviceName: weighted-routing
          servicePort: use-annotation
```

The one-line json for 'weighted-routing' above expands to:

```json
{ 
   "Type":"forward",
   "ForwardConfig":{ 
      "TargetGroupStickinessConfig":{ 
         "Enabled":"True",
         "DurationSeconds":100
      },
      "TargetGroups":[ 
         { 
            "Weight":20,
            "ServiceName":"service1",
            "ServicePort":"80"
         },
         { 
            "Weight":80,
            "TargetGroupArn":"arn:aws:******/external-legecy-tg"
         }
      ]
   }
}
```

The above splits traffic between a Kubernetes Service, and an **_external-provisioned target group_**, `arn:aws:******/external-legecy-tg`.

I'm unsure if it would be possible to split between two Kubernetes services. If I interpret the following message correctly, it may not be possible:

> The TargetGroups within ForwardConfig have been extended to allow use both Kubernetes Service and TargetGroupArn. Customers can configure backend with Kubernetes Service only / TargetGroupArn only / mix of two(for migration use case)

**Comment by jessesuen** [comment]
Confirmed from implementor that it can target 2+ services:

jessesuen:argo: Today at 11:28 AM
@M00nF1sh - does the alb.ingress.kubernetes.io/actions.weighted-routing annotation support the ability to split traffic between two kubernetes services? Or is it only Kubernetes + External ARN?
4 replies

M00nF1sh  15 minutes ago
it supports two kubernetes service(or more) :smile:

M00nF1sh  14 minutes ago
basically u can use either ServiceName+ServicePort or ExternalArn in the targetGroups array

**Comment by moensch** [comment]
Since Rollouts only modifies kubernetes `Service` objects during canary and doesn't touch AWS Target Groups, would we only support the ServiceName/ServicePort style of TargetGroup definition and not handle TargetGroupArn?
